src/main/scala/com/codacy/duplication/pmd/Cpd.scala

```diff
@@ -0,0 +1,154 @@
+    errStream: PrintStream): List[DuplicationClone] = {
+
+    languages.flatMap { lang =>
+      val configuration = getConfiguration(lang, options)
+      runWithConfiguration(configuration, directory)(errStream)
+    }
+
+  }
+
+  private def runWithConfiguration(config: pmd.cpd.CPDConfiguration, directory: Path)(
+    errStream: PrintStream): List[DuplicationClone] = {
+    val cpd = new pmd.cpd.CPD(config)
+    cpd.addRecursively(directory.toFile)
+
+    System.setErr(errStream)
+
+    cpd.go()
+
+    val res: util.Iterator[Match] = cpd.getMatches
+    res.asScala.map(matchToClone(_, directory)).toList
+  }
+
+  private def getLanguages(language: Option[Language]): Try[List[(pmd.cpd.Language, Int)]] = {
+
+    val languagesOpt: Option[Try[List[(cpd.Language, Int)]]] =
+      language.map { lang =>
+        convertLanguageToConfig(lang) match {
+          case Some(config) => Success(List(config))
+          case None =>
+            Failure(new Throwable(s"Cpd.getLanguages: there is no language with key $lang"))
+        }
+      }
+
+    languagesOpt.getOrElse(Success(allLanguages.flatMap(convertLanguageToConfig)))
+  }
+
+  private def convertLanguageToConfig(language: Language): Option[(pmd.cpd.Language, Int)] = {
+    language match {
+      case Languages.Python     => Some((new pmd.cpd.PythonLanguage, 50))
+      case Languages.Ruby       => Some((new pmd.cpd.RubyLanguage, 50))
+      case Languages.Java       => Some((new pmd.cpd.JavaLanguage, 100))
+      case Languages.Javascript => Some((new pmd.cpd.EcmascriptLanguage, 40))
+      case Languages.Scala =>
+        val cpdScala = new pmd.cpd.ScalaLanguage
+        object ScalaL
+            extends pmd.cpd.AbstractLanguage(
+              cpdScala.getName,
+              cpdScala.getTerseName,
+              ScalaTokenizer,
+              cpdScala.getExtensions.asScala: _*)
+        Some((ScalaL, 50))
+      case Languages.CSharp => Some((new pmd.cpd.CsLanguage, 50))
+      case _                => None
+    }
+  }
+
+  private def getConfiguration(language: (pmd.cpd.Language, Int),
+                               options: Map[DuplicationConfiguration.Key, DuplicationConfiguration.Value]) = {
+
+    val ignoreAnnotations = true
+    val skipLexicalErrors = true
+
+    val (cpdLanguage, defaultMinToken) = language
+
+    val minTokenMatch: Int = options.get(DuplicationConfiguration.Key("minTokenMatch")).fold(defaultMinToken) {
+      value: DuplicationConfiguration.Value =>
+        Option(value: JsValue).collect {
+          case JsNumber(number) => number.toInt
+          case _                => defaultMinToken
+        }.getOrElse(defaultMinToken)
+    }
+
+    val cfg = new pmd.cpd.CPDConfiguration()
+    cfg.setLanguage(cpdLanguage)
+    cfg.setIgnoreAnnotations(ignoreAnnotations)
+    cfg.setSkipLexicalErrors(skipLexicalErrors)
+    cfg.setMinimumTileSize(minTokenMatch)
+    cfg
+  }
+
+  private def matchToClone(m: pmd.cpd.Match, rootDirectory: Path): DuplicationClone = {
```

Comment:
No need to have the "matchTo" on the function name. Maybe rename to "duplicationClone"